src/linear_algebra/projective_space/subspace.lean

```diff
@@ -0,0 +1,162 @@
+
+lemma mem_add (T : subspace K V) (v w : V) (hv : v ≠ 0) (hw : w ≠ 0) (hvw : v + w ≠ 0) :
+  projectivization.mk K v hv ∈ T → projectivization.mk K w hw ∈ T →
+  projectivization.mk K (v + w) (hvw) ∈ T :=
+  T.mem_add' v w hv hw hvw
+
+/-- The span of a set of points in projective space is defined inductively to be the set of points
+which contains the original set, and contains all points determined by the (nonzero) sum of two
+nonzero vectors, each of which determine points in the span. -/
+inductive span_carrier (S : set (ℙ K V)) : set (ℙ K V)
+| of (x : ℙ K V) (hx : x ∈ S) : span_carrier x
+| mem_add (v w : V) (hv : v ≠ 0) (hw : w ≠ 0) (hvw : v + w ≠ 0) :
+    span_carrier (projectivization.mk K v hv) → span_carrier (projectivization.mk K w hw) →
+    span_carrier (projectivization.mk K (v + w) (hvw))
+
+/-- The span of a set of points in projective space is a subspace. -/
+def span (S : set (ℙ K V)) : subspace K V :=
+{ carrier := span_carrier S,
+  mem_add' := λ v w hv hw hvw,
+    span_carrier.mem_add v w hv hw hvw }
+
+/-- The span of a set of points contains the set of points. -/
+lemma subset_span (S : set (ℙ K V)) : S ⊆ span S := λ x hx, span_carrier.of _ hx
+
+/-- The span of a subspace is the subspace. -/
+@[simp] lemma span_coe (W : subspace K V) : span ↑W = W :=
+begin
+  ext, refine ⟨λ hx, _, λ hx, _⟩,
+  { induction hx with a ha u w hu hw huw _ _ hum hwm,
+      { exact ha },
+      { exact mem_add W u w hu hw huw hum hwm } },
+  { exact subset_span W hx },
+end
+
+/-- The span of a set of points is a Galois insertion between sets of points of a projective space
+and subspaces of the projective space. -/
+def gi : galois_insertion (span : set (ℙ K V) → subspace K V) coe :=
+{ choice := λ S hS, span S,
+  gc := λ A B, ⟨λ h, le_trans (subset_span _) h, begin
+    intros h x hx,
+    induction hx,
+    { apply h, assumption },
+    { apply B.mem_add, assumption' }
+  end⟩,
+  le_l_u := λ S, subset_span _,
+  choice_eq := λ _ _, rfl }
+
+/-- The ambient projective space is the top of the lattice of subspaces. -/
+instance has_top : has_top (subspace K V) :=
+⟨⟨⊤, λ _ _ _ _ _ _ _, trivial⟩⟩
+
+instance subspace_inhabited : inhabited (subspace K V) :=
+{ default := ⊤ }
+
+/-- The empty set is the bottom of the lattice of subspaces. -/
+instance has_bot : has_bot (subspace K V) :=
+⟨⟨⊥, λ v w hv hw hvw h, h.elim⟩⟩
+
+/-- The infimum of two subspaces exists. -/
+instance has_inf : has_inf (subspace K V) :=
+⟨λ A B, ⟨A ⊓ B, λ v w hv hw hvw h1 h2,
+  ⟨A.mem_add _ _ hv hw _ h1.1 h2.1, B.mem_add _ _ hv hw _ h1.2 h2.2⟩⟩⟩
+
+/-- Infimums of arbitrary collections of subspaces exist. -/
+instance has_Inf : has_Inf (subspace K V) :=
+⟨λ A, ⟨Inf (coe '' A), λ v w hv hw hvw h1 h2 t, begin
+  rintro ⟨s, hs, rfl⟩,
+  exact s.mem_add v w hv hw _ (h1 s ⟨s, hs, rfl⟩) (h2 s ⟨s, hs, rfl⟩),
+end⟩⟩
+
+/-- The subspaces of a projective space form a complete lattice. -/
+instance : complete_lattice (subspace K V) :=
```

Comment:
I may be wrong, but it seems like your definitions of `sup` and `Sup` are definitionally equal to those you'd get out of [`complete_lattice_of_Inf`](https://leanprover-community.github.io/mathlib_docs/order/complete_lattice.html#complete_lattice_of_Inf). You might want to consider using that as a shortcut.